### PR TITLE
[OPT P2] add inputs_embeds and cache

### PR DESCRIPTION
**Pull request by 1649759610**
### PR types

Function optimization

### PR changes

Models

### Description

1. support inputs_embeds and cache as input
2. add corresponding tests.
3. fix input_type bug in OPTEmbeddings.

### CODE REVIEW

File: tests/transformers/opt/test_modeling.py [hunk 1]
```diff
@@ -94,7 +93,13 @@ def prepare_config_and_inputs(self):
 
         input_mask = None
         if self.use_input_mask:
-            input_mask = random_attention_mask([self.batch_size, self.seq_length], dtype="int64")
+            # input_mask = random_attention_mask([self.batch_size, self.seq_length], dtype="int64")
```
Comment (sijunhe) [inline tests/transformers/opt/test_modeling.py]: rm comment

File: tests/transformers/opt/test_modeling.py [hunk 2]
```diff
@@ -260,7 +265,13 @@ def create_and_check_opt_model_past_large_inputs(self, config, input_ids, input_
 
         # create hypothetical next token and extent to next_input_ids
         next_tokens = ids_tensor((self.batch_size, 3), config.vocab_size, dtype="int64")
-        next_mask = ids_tensor((self.batch_size, 3), vocab_size=2, dtype="int64")
+        # next_mask = ids_tensor((self.batch_size, 3), vocab_size=2, dtype="int64")
```
Comment (sijunhe) [inline tests/transformers/opt/test_modeling.py]: rm comment

File: tests/transformers/opt/test_modeling.py [hunk 3]
```diff
@@ -94,7 +93,13 @@ def prepare_config_and_inputs(self):
 
         input_mask = None
         if self.use_input_mask:
-            input_mask = random_attention_mask([self.batch_size, self.seq_length], dtype="int64")
+            # input_mask = random_attention_mask([self.batch_size, self.seq_length], dtype="int64")
+            input_mask_cond = paddle.randn([self.batch_size, self.seq_length])
+            input_mask = paddle.where(
```
Comment (sijunhe) [inline tests/transformers/opt/test_modeling.py:98]: 这里看起来也是一个随机的mask，只是是0 或者 -1e4?

File: tests/transformers/opt/test_modeling.py (hunk 1 above)
Comment (1649759610) [reply tests/transformers/opt/test_modeling.py]: 已删除

File: tests/transformers/opt/test_modeling.py (hunk 2 above)
Comment (1649759610) [reply tests/transformers/opt/test_modeling.py]: 已删除

File: tests/transformers/opt/test_modeling.py (hunk 3 above)
Comment (1649759610) [reply tests/transformers/opt/test_modeling.py:98]: 当前需要这样填充： 0：unmasked token, -1e4: masked token

关于attention_mask的注释，最开始写的-1e-9，但应该是有问题，应该为-1e9以适配model.generate， 故而注释中改成了-1e9。

然而原始代码中用的-1e4，考虑到-1e4和-1e9对于计算attention差别不大，所以未做进一步改动。

**Review by sijunhe — APPROVED** [review]
lgtm!
